src/context/CategoryContext.js

```diff
@@ -0,0 +1,28 @@
+import axios from "axios";
+import React from "react";
+import { createContext, useContext, useEffect, useState } from "react";
+
+const CategoryContext = createContext();
+const useCategoryContext = () => useContext(CategoryContext);
+
+const CategoryContextProvider = ({ children }) => {
+  const [categoryList, setCategoryList] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        const data = await axios.get("/api/categories");
+        console.log(data.data.categories);
+        setCategoryList(data.data.categories);
+      } catch (err) {
+        console.log(err);
```

Comment:
here you can alert the error  or maybe you can create one state for error and show that on UI

Reply:
Alright, I will try to do it. Thank You.